### ISSUE TITLE
Add performance tests for C++ plugin in current model

### ISSUE THREAD

**Issue by big-guy** [gradle/gradle-native#158]
Our existing native performance tests use the software model.

We should have coverage for:

- Single project of a "reasonable" size
- Multi-project build with dependencies between projects
- Effect of build cache

**Comment by eljobe** [comment]
I wonder if we can break the scope of this card into a few cards. @wolfs was mentioning that he needed the coverage currently provided by the software model's performance tests to unblock his work on discovered inputs. I think we should have a high-priority issue to add the equivalent performance tests for the current model plugins and a separate issue or maybe a few separate issues for performance tests which cover other aspects of our native plugins. Thoughts?

**Comment by big-guy** [comment]
@eljobe left you a couple of comments

**Comment by eljobe** [comment]
I made some really good progress on adding the project-project dependency generation needed to implement multi-project cpp builds today. I'm hopeful that I'll have something ready before our iteration ends tomorrow night.
